File: KHost.App/ClientApp/src/app/modules/karaoke/models/QueuedSong.ts
import { Singer } from "../../../models/Singer";
import { Song } from "src/app/models/Song";

export class QueuedSong
{
    id: number = -1;
    songId: number | null = null;
    singerId: number | null = null;

    song: Song | null = null;
    singer: Singer | null = null;
}

File: KHost.App/Models/QueuedSong.cs
namespace KHost.App.Models
{
    public class QueuedSong : BaseModel, IModelWithPosition, IModelWithId
    {
        public int? Id { get; set; }

        public int SongId { get; set; }

        public int QueuedSingerId { get; set; }

        public int Position { get; set; }
    }
}
